File: length_after_transformations/length_after_transformations.py
class Solution:
  def lengthAfterTransformations(self, s: str, t: int) -> int:
    MOD = 10**9 + 7
    count = [0] * 26
    for ch in s:
      count[ord(ch) - ord('a')] += 1

    for _ in range(t):
      new_count = [0] * 26
      for i in range(26):
        if i == 25: 
          new_count[0] = (new_count[0] + count[i]) % MOD 
          new_count[1] = (new_count[1] + count[i]) % MOD 
        else:
          new_count[i + 1] = (new_count[i + 1] + count[i]) % MOD
      count = new_count

    return sum(count) % MOD
      
        
s = Solution()
print(s.lengthAfterTransformations("abcyy", 2))  
print(s.lengthAfterTransformations("azbk", 1))   

File: length_after_transformations/length_after_transformations.ts
class Solution {
  lengthAfterTransformations(s: string, t: number): number {
    const MOD = 1_000_000_007;
    let count = new Array(26).fill(0);
    for (let ch of s) {
      count[ch.charCodeAt(0) - 97]++;
    }

    for (let step = 0; step < t; step++) {
      const newCount = new Array(26).fill(0);
      for (let i = 0; i < 26; i++) {
        if (i === 25) {
          newCount[0] = (newCount[0] + count[i]) % MOD;
          newCount[1] = (newCount[1] + count[i]) % MOD;
        } else {
          newCount[i + 1] = (newCount[i + 1] + count[i]) % MOD;
        }
      }
      count = newCount;
    }

    return count.reduce((acc, val) => (acc + val) % MOD, 0);
  }
}

const s = new Solution();
console.log(s.lengthAfterTransformations("abcyy", 2)); // 7
console.log(s.lengthAfterTransformations("azbk", 1));  // 5
